packages/orchestrator-ui-components/src/components/WfoForms/formFields/IpPrefixTable.tsx

```diff
@@ -0,0 +1,453 @@
+            ? filteredIpBlocks.reverse()
+            : filteredIpBlocks;
+    }
+
+    selectPrefix = (prefix: IpBlock) => () => {
+        if (prefix.state === 0 || prefix.state === 1) {
+            this.setState({ selectionDone: true });
+            this.props.onChange(prefix);
+        }
+    };
+
+    render() {
+        const { id, name, selected_prefix_id } = this.props;
+        const {
+            filteredPrefixes,
+            manualOverrideVisible,
+            selectionDone,
+            loading,
+            manualOverrideValue,
+        } = this.state;
+        const ipBlocks = this.filterAndSortBlocks();
+
+        const columns = [
+            {
+                field: 'id',
+                name: 'ID',
+                sortable: true,
+                truncateText: true,
+            },
+            {
+                field: 'prefix',
+                name: 'Prefix',
+                sortable: true,
+            },
+            {
+                field: 'description',
+                name: 'Description',
+                truncateText: true,
+                sortable: true,
+            },
+            {
+                field: 'state',
+                name: 'Status',
+                sortable: true,
+                render: (prefixState: number) => (
+                    <EuiBadge color={prefixState === 3 ? 'danger' : 'success'}>
+                        {ipamStates[prefixState]}
+                    </EuiBadge>
+                ),
+            },
+            {
+                field: 'Action',
+                name: '',
+                render: (id: string, record: IpBlock) =>
+                    record.state !== 3 ? (
+                        <EuiButton
+                            onClick={this.selectPrefix(record)}
+                            id={`select-prefix-${id}-button`}
+                        >
+                            Select
+                        </EuiButton>
+                    ) : null,
+            },
+        ];
+
+        const { state, prefix } = { ...this.state.filter };
+        const parentPrefix = prefix?.id;
+
+        const options: Option[] = filteredPrefixes.map((fp) => ({
+            value: fp.id.toString(),
+            label: fp.prefix,
+        }));
+        const value = options.find(
+            (option) => option.value === parentPrefix?.toString(),
+        );
+
+        const pagination = {
+            initialPageSize: 25,
+            pageSizeOptions: [25, 50, 100, 250],
+        };
+
+        if (selected_prefix_id && selectionDone) {
+            return (
+                <EuiButton
+                    id="undo-parent-prefix-choice-button"
+                    iconType="editorUndo"
+                    onClick={() => this.setState({ selectionDone: false })}
+                >
+                    Reset parent prefix choice
+                </EuiButton>
+            );
+        }
+        // const customStyles = getReactSelectTheme(this.context.theme);
```

Comment:
commented code

Reply:
No.

This would involve: ClassBased -> Function based.

const customStyles = getReactSelectTheme(this.context.theme); -> but the context is already used for the ApiClient

Created #482